x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/create_query_alert_type.ts

```diff
@@ -32,6 +32,18 @@ export const createQueryAlertType = (
           }
           return validated;
         },
+        /**
+         * validate rule params when rule is updated or created
+         * returned params can be modified (useful in case of version increment)
+         * @param mutatedRuleParams
+         * @returns mutatedRuleParams
+         */
+        validateMutatedParams: (mutatedRuleParams) => {
+          validateImmutable(mutatedRuleParams.immutable);
```

Comment:
When we install new prebuilt rules we do it via calling `RulesClient.create`, obviously `params.immutable` == `true` in this case. Now, `validateImmutable` throws when `params.immutable` == `true`, but the comment says "validate rule params when rule is updated or **created**". So does RulesClient call `validateMutatedParams` when a rule is created? Loading prebuilt rules works w/o errors in this branch.

Reply:
Corrected a bit description. It applied on bulkEdit for now, and not on update/create yet